platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ExportMigrationManagerImpl.java

```diff
@@ -160,16 +174,18 @@ public void doExport(String productBranding, String productVersion) {
   public void close() throws IOException {
     if (!closed) {
       this.closed = true;
+      CipherOutputStream cos = cipherUtils.getCipherOutputStream(zipOutputStream);
       try {
         zipOutputStream.closeEntry();
         try {
           zipOutputStream.putNextEntry(
               new ZipEntry(MigrationContextImpl.METADATA_FILENAME.toString()));
-          JsonUtils.MAPPER.writeValue(zipOutputStream, metadata);
+          JsonUtils.MAPPER.writeValue(cos, metadata);
         } catch (IOException e) {
           throw new MigrationException(Messages.EXPORT_METADATA_CREATE_ERROR, e);
         }
       } finally {
+        cos.close();
```

Comment:
❗️ 🤔  Not sure we should be closing this one. If you need to because of the cipher then we should wrap the output stream passed to the cipher output stream using the CloseShieldOutputStream class. If you check above, we created a ProxyOutputStream to protect the zip stream closing. The next line will properly close the entry and the zip stream.